keyboards/ploopyco/trackball_thumb/info.json

```diff
@@ -18,7 +18,12 @@
         "mousekey": true,
         "nkro": true,
         "pointing_device": true,
-        "encoder": false
+        "encoder": true
+    },
+    "encoder": {
+        "rotary": [
+            {"pin_a": "NO_PIN", "pin_b": "NO_PIN"}
```

Comment:
I think padding of pin config for custom matrix so it sets matrix size, was something we have said we want to avoid.

If this is the case here, should we not just advertise something specific for "custom" than dirty the concept of "rotary"?

Reply:
So the intention here seems to be that a pure "custom" driver wouldnt reuse the `rotary` object. The usage of `NO_PIN` would be where the implementation is mainly core, but with just the init/read functions overridden. 

For that scenario, using `NO_PIN` seems to match existing patterns and would be fine with me (at least till we potentially tackle that greater issue).

Reply:
Correct -- for the current graycode-style encoders we'd see:
```json
{
	"encoder": {
		/* "driver": "graycode", */ // this is implicitly present if left unspecified
		"rotary": [
			/* list of encoders, for now we'd allow for `null` or `NO_PIN` here as per other configs */
		]
	}
}
```

Otherwise, for a custom driver, we'd see:
```json
{
	"encoder": {
		"driver": "custom"
	}
}
```

Reply:
All ploopy's converted over to setting custom driver in `info.json`, manually `#define NUM_ENCODERS 1` in respective `config.h` files.